src/components/AutocompleteSearchBar/AutocompleteSearchBar.js

```diff
@@ -0,0 +1,150 @@
+ *
+ * React-SearchKit is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Input } from 'semantic-ui-react';
+
+class AutocompleteSearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.updateQueryString = this.props.updateQueryString;
+    this.updateSuggetions = this.props.debounce
+      ? _.debounce(this.props.updateSuggetions, this.props.debounceTime, {
+          leading: true,
+        })
+      : this.props.updateSuggetions;
+    this.clearSuggestions = this.props.clearSuggestions;
+    this.state = {
+      currentValue: this.props.queryString || '',
+    };
+    this.renderElement = props.renderElement || this._renderElement;
+    this.renderSuggestions = props.renderSuggestions || this._renderSuggestions;
+    this.handleAutocompleteChange =
+      props.handleAutocompleteChange || this._handleAutocompleteChange;
+  }
+
+  _renderSuggestions(querySuggestions) {
+    // Define actions
+    const onSuggestionSelected = async suggestion => {
+      await this.setState({
+        currentValue: suggestion,
+      });
+      this.clearSuggestions();
+      this.executeSearch();
+    };
+
+    //Render Element
+    if (querySuggestions.length === 0) {
+      return null;
+    }
+
+    return (
+      <ul>
+        {querySuggestions.map(text => (
+          <li onClick={() => onSuggestionSelected(text)} key={text}>
+            {text}
+          </li>
+        ))}
+      </ul>
+    );
+  }
+
+  _renderElement(
+    placeholder,
+    queryString,
+    querySuggestions,
+    onInputChange,
+    executeSearch
+  ) {
+    // Initialize values
+    placeholder = placeholder || 'Type something';
+    // Define actions
+    const onBtnSearchClick = (event, input) => {
+      executeSearch();
+    };
+    const onKeyPress = (event, input) => {
+      if (event.key === 'Enter') {
+        executeSearch();
+      }
+    };
+
+    //Render Element
+    return (
+      <div className="AutoCompleteText">
+        <Input
+          action={{
+            content: 'Search',
+            onClick: onBtnSearchClick,
+          }}
+          fluid
+          placeholder={placeholder}
+          onChange={(event, { value }) => {
+            onInputChange(value);
+          }}
+          value={queryString}
+          onKeyPress={onKeyPress}
+        />
+        {this.renderSuggestions(querySuggestions)}
+      </div>
+    );
+  }
+
+  _handleAutocompleteChange = () => {
+    if (this.state.currentValue.length >= 3) {
+      this.updateSuggetions(this.state.currentValue);
+    }
+  };
+
+  onInputChange = async queryString => {
+    await this.setState({
+      currentValue: queryString,
+    });
+
+    this.handleAutocompleteChange();
+  };
+
+  executeSearch = () => {
+    this.updateQueryString(this.state.currentValue);
+  };
+
```

Comment:
I just had an order issue in the method declarations, probably my ocd :P, could you move the `_renderElement` here?

Reply:
done